### PR TITLE
librustc_errors: Add some more documentation

### PR DESCRIPTION
**Pull request by phansch**
r? @estebank

### CODE REVIEW

File: src/librustc_errors/lib.rs [hunk 1]
```diff
@@ -1,3 +1,7 @@
+//! Diagnostics creation and emission for rustc
```
Comment (Centril) [inline src/librustc_errors/lib.rs]:
```suggestion
//! Diagnostics creation and emission for `rustc`.
```

File: src/librustc_errors/diagnostic_builder.rs [hunk 2]
```diff
@@ -348,7 +348,7 @@ impl<'a> DiagnosticBuilder<'a> {
 
     /// Convenience function for internal use, clients should use one of the
     /// struct_* methods on Handler.
-    pub fn new_with_code(handler: &'a Handler,
+    pub(crate) fn new_with_code(handler: &'a Handler,
```
Comment (Centril) [inline src/librustc_errors/diagnostic_builder.rs]:
```suggestion
    crate fn new_with_code(handler: &'a Handler,
```

File: src/librustc/session/config.rs [hunk 3]
```diff
@@ -227,13 +227,17 @@ impl OutputType {
     }
 }
 
+/// The type of diagnostics output to generate
```
Comment (Centril) [inline src/librustc/session/config.rs]:
```suggestion
/// The type of diagnostics output to generate.
```

File: src/librustc/session/config.rs [hunk 4]
```diff
@@ -227,13 +227,17 @@ impl OutputType {
...
 #[derive(Clone, Copy, Debug, PartialEq, Eq)]
 pub enum ErrorOutputType {
+    /// Output meant for the consumption of humans.
     HumanReadable(HumanReadableErrorType),
+    /// Output that's consumed by other tools such as `rustfix` or the `RLS`.
     Json {
-        /// Render the json in a human readable way (with indents and newlines)
+        /// Render the JSON in a human readable way (with indents and newlines)
```
Comment (Centril) [inline src/librustc/session/config.rs]:
```suggestion
        /// Render the JSON in a human readable way (with indents and newlines).
```

**Review by Centril — CHANGES_REQUESTED** [review]
Mostly proof reading :)

**Comment by estebank** [comment]
R=me with the comments addressed

**Pull request merged**